api/routes.go

```diff
@@ -39,6 +39,9 @@ func (s *Server) RegisterRoutes() {
 	r.Combo("/index/list", ready, bind(models.IndexList{})).Get(s.indexList).Post(s.indexList)
 	r.Combo("/index/delete", ready, bind(models.IndexDelete{})).Get(s.indexDelete).Post(s.indexDelete)
 	r.Combo("/index/get", ready, bind(models.IndexGet{})).Get(s.indexGet).Post(s.indexGet)
+	r.Combo("/index/tags", ready, bind(models.IndexTags{})).Get(s.indexTags).Post(s.indexTags)
+	r.Combo("/index/tags/findSeries", ready, bind(models.IndexTagFindSeries{})).Get(s.indexTagFindSeries).Post(s.indexTagFindSeries)
+	r.Combo("/index/tags/:tag([0-9a-zA-Z]+)", ready, bind(models.IndexTagDetails{})).Get(s.indexTagDetails).Post(s.indexTagDetails)
```

Comment:
does it make sense to have a subhierachy here? 
this is meant to be a 1:1 mapping of http url to index operation, also naming wise, so I think this would make more sense:
```
"/index/tags" -> s.indexTags
"/index/find_by_tag"-> s.indexFindByTag
"/index/tag_details" -> s.indexTagDetails // specify the tag via the model, not via the url
```

Reply:
When I named them I basically took the urls as described in the graphite docs and pre-pended `/index` because those are all the internal calls to only be used by other cluster: nodes.http://graphite.readthedocs.io/en/latest/tags.html#exploring-tags
I'd actually like the urls you suggest more as well, but I think that would be unnecessarily confusing or not?

Reply:
personally I like the 1:1 mapping of http path to index operation, and the more simple we can keep our patterns and naming of functions etc, the easier this becomes to read IMHO.

Reply:
This needs to match the graphite url hierarchy so that it is queryable by the graphite http tagdb

Reply:
@DanCech actually these urls are only used for communication between cluster peers, to fanout a query that came in. But the reason why I named them the way they are now is because that makes it easy to figure out which of those internal endpoints is used by which of the graphite endpoints.

Reply:
> the reason why I named them the way they are now is because that makes it easy to figure out which of those internal endpoints is used by which of the graphite endpoints.

this won't work well anymore I think when these endpoints start to get used for other things. for example in the future MT will start hitting some of these endpoints as it is responding to render requests. that's why i think it's more future proof that an endpoint http path represents the operation that will be performed. also for diagnosing issues i think it makes morse sense.

Reply:
ok, renaming

Reply:
https://github.com/grafana/metrictank/pull/750/commits/d30bbe7655357a99325da08b607d45e088b0ce28

Reply:
why is all the stuff in between the `find_by_tag` route and the `s.MetricIndex.FindByTag()` call, still called `indexTagFindSeries` ? (both the function, and input and output models)

Reply:
ping @replay ^^